src/classes/speciestorsion.cpp

```diff
@@ -378,6 +379,21 @@ double SpeciesTorsion::energy(double angleInDegrees, int form, const std::vector
          */
         return 0.5 * params[0] * (1.0 - cos(params[1] * params[2] / DEGRAD) * cos(params[1] * phi));
     }
+    else if (form == SpeciesTorsion::FourierNForm)
+    {
+        /*
+         * U(phi) = k * (C0 + C1 cos(phi) + C2 cos(2*phi) ... Cn cos(n*phi))
+         *
+         * Parameters:
+         * 0 : Force constant, k
+         * 1...N : Coefficients C0 - CN
+         */
+        auto U = params[0] * params[1];
+        for (auto n = 2; n < params.size(); ++n)
+            U += params[0] * params[n] * cos((n - 1) * phi);
+
+        return U;
```

Comment:
```suggestion
        std::vector<int> ns(params.size);
        std::iota(ns.begin(), ns.end(), 1);
        return params[0] * std::transform_reduce(params.begin(), params.end(), ns.begin(), params[1], std::plus<>(), [phi](auto &param, auto &n) {return param * (1.0 + cos(n * phi));});
```

With same caveats as above.